tokio/src/runtime/task/join.rs

```diff
@@ -45,6 +46,49 @@ doc_rt_core! {
     /// # }
     /// ```
     ///
+    /// Explicit JoinHandle<T> specification for a spawned task that does not return a value.
+    /// This is parameterized over `()`:
+    ///
+    /// ```
+    /// use tokio::task;
+    ///
+    /// # async fn doc() {
+    /// let join_handle: task::JoinHandle<()> = task::spawn(async {
+    ///     println!("I return nothing.")
+    /// });
+    /// # }
+    /// ```
+    ///
+    /// Explicit JoinHandle<T> specification for a spawned task that returns a value:
+    ///
+    /// ```
+    /// use tokio::task;
+    ///
+    /// # async fn doc() {
+    /// let join_handle: task::JoinHandle<i32> = task::spawn(async {
+    ///     5 + 3
+    /// });
+    /// # }
+    ///
+    /// ```
+    ///
+    /// Returning a `Result` from a spawned task. Note the double chaining of the `?`
+    /// operator to match onto the `Ok` value (as, the return value is a
+    /// `Result<Result<i32, &str>, JoinError>`).
+    ///
+    /// ```
+    /// use tokio::task;
+    ///
+    /// # async fn doc() -> Result<(), Box<dyn std::error::Error>> {
+    /// let join_handle: task::JoinHandle<Result<i32, &str>> = task::spawn(async {
+    ///     Ok(5 + 3)
+    /// } );
+    ///
+    /// let result = join_handle.await??;
+    /// # Ok(())
+    /// # }
```

Comment:
In this case I think it is better to include the function in the example such that the return value is visible.
```suggestion
    /// use std::io;
    /// use tokio::task::JoinHandle;
    ///
    /// #[tokio::main]
    /// async fn main() -> io::Result<()> {
    ///     let join_handle: JoinHandle<Result<i32, &str>> = tokio::spawn(async {
    ///         Ok(5 + 3)
    ///     });
    ///
    ///     let result = join_handle.await??;
    ///     assert_eq!(result, 8);
    ///     Ok(())
    /// }
```